src/wazuh_modules/agent_upgrade/agent/wm_agent_upgrade_agent.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Wazuh Module for Agent Upgrading
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * July 30, 2020.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "wazuh_modules/wmodules.h"
+#include "wm_agent_upgrade_agent.h"
+
+const char* upgrade_values[] = {
+    [WM_UPGRADE_SUCCESSFULL] = "0",
+    [WM_UPGRADE_FAILED] = "2"
+};
+
+const char* upgrade_messages[] = {
+    [WM_UPGRADE_SUCCESSFULL] = "Upgrade was successfull",
+    [WM_UPGRADE_FAILED]      = "Upgrade failed"
+};
+
+/**
+ * Reads the upgrade_result file if it is present and sends the upgrade result message to the manager.
+ * Example message:
+ * {
+ *   "command": "agent_upgraded/agent_upgrade_failed",
+ *   "params":  {
+ *     "error": 0/{ERROR_CODE},
+ *     "message": "Upgrade was successfull"
+ *   }
+ * }
+ * @param queue_fd File descriptor of the upgrade queue
+ * @param state upgrade result state
+ * */
+static void wm_upgrade_agent_send_ack_message(int queue_fd, wm_upgrade_agent_state state);
+
+void wm_agent_upgrade_check_status() {
+    char buffer[20];
+    FILE * result_file;
+    const char * PATH = WM_AGENT_UPGRADE_RESULT_FILE;
+    /**
+     *  StartMQ will wait until agent connection which is when the pkg_install.sh will write 
+     *  the upgrade result
+    */
+    int queue_fd = StartMQ(DEFAULTQPATH, WRITE, MAX_OPENQ_ATTEMPS);
+
+    if (queue_fd < 0) {
+        mterror(WM_AGENT_UPGRADE_LOGTAG, WM_UPGRADE_QUEUE_FD);
+    } else {
+        if (result_file = fopen(PATH, "r"), result_file) {
+            fgets(buffer,20,result_file);
+            fclose(result_file);
+        }
+
+        wm_upgrade_agent_state state;
```

Comment:
If the result file doesn't exist, you shouldn't execute this loop.

Reply:
You are right